content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -138,6 +162,45 @@ the possible following fields:
 
 `dvc.yaml` files also support `# comments`.
 
+### dvc.lock file
+
+For every `dvc.yaml` file, a corresponding `dvc.lock` (YAML) file is created or
+updated by DVC as well. Its purpose is to lock the exact contents of all files
```

Comment:
it's not only about files I think

Reply:
it's not the only purpose - it also replaces .dvc files for pipeline outputs for example

Reply:
Also, it might lock the values of params, not just files.

Reply:
This has already been rewriten (PTAL) but it OK, what does dvc.lock lock to exactly? What's the broader concept? Saying "locks to the exact content of files, and  parameter key/value pairs" seems too specific for an introductory paragraph.

Reply:
I'm going to go with "pipeline state" for now.